### DISCUSSION TITLE
Problem with assigning issues to users

### DISCUSSION THREAD

**Discussion by MichaelSweikata** [python-gitlab/python-gitlab#2021] [Q&A] [answered]
## Description of the problem, including code/CLI snippet

Under the Gitlab API, in order to assign an issue to users, you must specify their user ID to the list `assignee_ids`. In the documentation and API reference, I don't see `assignee_ids`, I see `assignees`. If I manually set the attribute on the object, it works.

## Expected Behavior

```python
for index, assignees in enumerate(issue_assignees):
  assignees.assignee_ids.append(user_id)
  issue_assignees[index].save()
```
The ticket is assigned to a provided `user_id`.

## Actual Behavior
```python
issues = project.issues.list(state='opened')
for index, issue in enumerate(issues):
    if not issue.assignees:
      issue_assignees = get_group_members(stuff) //this is a custom function that I give it a group name, and it returns the IDs of the users in that group in a list e.g., [20, 40, 50, 21, 25]
    t = []
    for assignees in issue_assignees:
       t.append(assignees.id)
         issue.assignee_ids = t
       issues[index].save() //on this line the issue at position index refreshes with no assignees
```
And the ticket is not updated, nor is there a return code of an issue.

## Specifications

  - python-gitlab version: 3.4.0
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): 14.9.3-ee

**Comment by JohnVillalovos** [comment]
Can you provide more context please? How was `issue_assignees` created?

**Comment by MichaelSweikata** [comment]
Sorry, I'll add/update a more complete version to the report.

**Comment by JohnVillalovos** [comment, marked as answer]
Maybe try something like this?

```python
issues = project.issues.list(state='opened')
for index, issue in enumerate(issues):
    issue = project.issues.get(issue.id)  # This may not be necessary but usually a good idea.
    if not issue.assignees:
      issue_assignees = get_group_members(stuff) //this is a custom function that I give it a group name, and it returns the IDs of the users in that group in a list e.g., [20, 40, 50, 21, 25]
    else:
        continue
    t = []
    for assignees in issue_assignees:
       t.append(assignees.id)
    issue.assignee_ids = t
    issue.save()
    issue = project.issues.get(issue.id)
    issue.pprint()
```

**Reply by MichaelSweikata** [reply]
Ultimately I was able to make it work by doing that, but I don't see `assignee_ids` in the documentation. I checked: https://python-gitlab.readthedocs.io/en/stable/api/gitlab.v4.html#gitlab.v4.objects.ProjectIssueManager
and it lists `assignee_id`, which I believe was deprecated in favor of `assignee_ids`. So, when I just extended `issue` and added `assignee_ids`, it did work.

**Reply by JohnVillalovos** [reply]
Great to hear you got it working! 

So it is documented here: https://docs.gitlab.com/ee/api/issues.html#edit-issue

If you would like to do a PR to help improve the `python-gitlab` documentation it would be appreciated. This area could use some help: https://python-gitlab.readthedocs.io/en/stable/gl_objects/issues.html

As a note it appears that `assignee_ids` is not returned in the `get()` but is used when creating or editing an issue.

**Reply by MichaelSweikata** [reply]
Okay, so that actually helps me understand the docs _a little better_. Because essentially what you've got is any time Gitlab makes a change to that api, you're automatically adjusting for it by letting the user add new parameters on the fly.

But yes, I'm happy to contribute to the project in that way. I'll get a branch together for review.

**Reply by MichaelSweikata** [reply]
So I have some documentation written out following the contributors guide, but I don't have permission to push a branch up.

**Reply by nejch** [reply]
@MichaelSweikata you can contribute from a fork :)
https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow
https://gist.github.com/Chaser324/ce0505fbed06b947d962

**Reply by MichaelSweikata** [reply]
Ah ha! Thank you. Haven't contributed this way before, so, thanks for pointing that out. I'll have to read up on Forking Workflows.

**Comment by MichaelSweikata** [comment]
Okay, apologies, I couldn't figure out the commit squashing, and rather than spin in circles for a few days while a simple doc update just needs to go in, I made a new branch with a single commit containing the changes:

https://github.com/python-gitlab/python-gitlab/pull/2025
